tooling/templatize/main.go

```diff
@@ -126,3 +130,34 @@ func main() {
 		log.Fatal(err)
 	}
 }
+
+func (opts *GenerationOptions) ExecuteTemplate(ctx context.Context) error {
```

Comment:
I'd think we need to add some test from the start for this. Can you please add at least some basic test case?